### ISSUE TITLE
Multicast Server "-S" ignores Interface Option "-I"

### ISSUE THREAD

**Issue by uwedisch** [knxd/knxd#479]
The multicast server "-S" ignores interface option "-I" with the next setup:
- Branch "deb", current commit
- Ubuntu 18.04 LTS
- "-I br0" on enp5s0
- "-S 224.0.23.12:3671"
- enp5s0.1 existing

It's expected that for multicast traffic outgoing interface "br0" ist used.  But this is ignored and with specific setup interface "enp5s0.1" is used (VLAN no. 1 on enp5s0 instead bridge on enp5s0).

This gives for example subsequent problems when routing multicast traffic with smcroute or the like.

**Comment by smurfix** [comment]
Complete command line / INI file and "ip addr" output please.

**Comment by uwedisch** [comment]
KNXD_OPTS="-e 3.2.40 -E 3.2.41:8 -D -T -R -S 224.0.23.12:3671 -I br0 -n 'knxd@linux' -u /tmp/eib -b tpuarts:/dev/ttyS0"

**Comment by uwedisch** [comment]
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp5s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master br0 state UP group default qlen 1000
    link/ether 14:da:e9:f4:b7:ff brd ff:ff:ff:ff:ff:ff
3: enp5s0.1@enp5s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 14:da:e9:f4:b7:ff brd ff:ff:ff:ff:ff:ff
    inet 192.168.100.100/24 brd 192.168.100.255 scope global enp5s0.1
       valid_lft forever preferred_lft forever
    inet6 fe80::16da:e9ff:fef4:b7ff/64 scope link 
       valid_lft forever preferred_lft forever
4: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 14:da:e9:f4:b7:ff brd ff:ff:ff:ff:ff:ff
    inet 192.168.101.100/24 brd 192.168.101.255 scope global br0
       valid_lft forever preferred_lft forever
    inet6 fe80::16da:e9ff:fef4:b7ff/64 scope link 
       valid_lft forever preferred_lft forever
5: anbox0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether a6:71:6f:95:a2:9c brd ff:ff:ff:ff:ff:ff
    inet 192.168.250.1/24 scope global anbox0
       valid_lft forever preferred_lft forever
    inet6 fe80::a471:6fff:fe95:a29c/64 scope link 
       valid_lft forever preferred_lft forever
6: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 52:54:00:3f:18:2e brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
       valid_lft forever preferred_lft forever
7: virbr0-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc fq_codel master virbr0 state DOWN group default qlen 1000
    link/ether 52:54:00:3f:18:2e brd ff:ff:ff:ff:ff:ff
8: vnet0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master br0 state UNKNOWN group default qlen 1000
    link/ether fe:54:00:a2:5a:86 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc54:ff:fea2:5a86/64 scope link 
       valid_lft forever preferred_lft forever
```

**Comment by smurfix** [comment]
Let's start with the obvious, which is that `-I` tries to modify `-R` and thus needs to appear before that option. The same thing applies to `-n`. Yes, knxd (or rather `knxd_args`) should probably catch that.

Also, if you have a bridged network, the individual interfaces attached to it really shouldn't have their own address, much less the same one as the interface.

**Comment by uwedisch** [comment]
Thank you, now I see partial traffic on routing protocol.  My change was:
`KNXD_OPTS="-e 3.2.40 -E 3.2.41:8 -c -D -T -I br0 -n 'knxd@kaleb' -R -S 224.0.23.12:3671 -u /tmp/eib -b tpuarts:/dev/ttyS0"`

I also changed network interface from `enp5s0.1@enp5s0` to `br0.1@br0`.

With tshark and a group read with ETS via routing protocol I see the read, but not the response (same with bus monitor of ETS via routing protocol):
`   10 13.433887725 192.168.101.100 → 224.0.23.12  KNXnetIP 61 ROUTING_INDICATION 3671 > 3671| cEMI: L_Data.ind`

But with parallel bus monitor of ETS via tunneling protocol I see group read and group read response if I do a group read via routing protocol.  My expectation is to see read response also with routing protocol.

`br0.1` is quiet with these tests.

**Comment by dallmair** [comment]
This has been fixed in commit 89a8285 and released in version 0.14.62.
